backends/graphics/atari/atari-graphics.h

```diff
@@ -0,0 +1,272 @@
+
+	static const int SCREENS = 3;
+	static const int FRONT_BUFFER = 0;
+	static const int BACK_BUFFER1 = 1;
+	static const int BACK_BUFFER2 = 2;
+	byte *_screen[SCREENS] = {};	// for Mfree() purposes only
+	byte *_screenAligned[SCREENS] = {};
+	Graphics::Surface _screenSurface;
+
+	byte *_chunkyBuffer = nullptr;	// for Mfree() purposes only
+	Graphics::Surface _chunkySurface;
+
+	byte *_overlayScreen = nullptr;	// for Mfree() purposes only
+	Graphics::Surface _screenOverlaySurface;
+
+	byte *_overlayBuffer = nullptr;	// for Mfree() purposes only
+	Graphics::Surface _overlaySurface;
+
+private:
+	enum CustomEventAction {
+		kActionToggleAspectRatioCorrection = 100,
+	};
+
+	void setVidelResolution() const;
+	void waitForVbl() const;
+
+	bool updateOverlay();
+	bool updateDirectBuffer();
+	bool updateSingleBuffer();
+	bool updateDoubleAndTripleBuffer();
+
+	virtual void* allocFast(size_t bytes) const = 0;
+
+	virtual void copySurfaceToSurface(const Graphics::Surface &srcSurface, Graphics::Surface &dstSurface) const = 0;
+	virtual void copyRectToSurface(const Graphics::Surface &srcSurface, int destX, int destY,
+								   Graphics::Surface &dstSurface, const Common::Rect &subRect) const = 0;
+	virtual void copyRectToSurfaceWithKey(const Graphics::Surface &srcSurface, int destX, int destY,
+										  Graphics::Surface &dstSurface, const Common::Rect &subRect, uint32 key) const = 0;
+	virtual void alignRect(const Graphics::Surface &srcSurface, Common::Rect &rect) const {}
+	virtual void sync() const {}
+
+	enum class ScaleMode {
+		None,
+		Upscale,
+		Downscale
+	};
+	void copySurface8ToSurface16(const Graphics::Surface &srcSurface, const byte *srcPalette,
+								 Graphics::Surface &dstSurface, int destX, int destY,
+								 const Common::Rect subRect, ScaleMode scaleMode) const;
+	void copySurface8ToSurface16WithKey(const Graphics::Surface &srcSurface, const byte* srcPalette,
+										Graphics::Surface &dstSurface, int destX, int destY,
+										const Common::Rect subRect, uint32 key) const;
+
+	void handleModifiedRect(const Graphics::Surface &surface, Common::Rect rect, Common::Array<Common::Rect> &rects) const;
+
+	void updateCursorRect();
+
+	bool _aspectRatioCorrection = false;
+	bool _oldAspectRatioCorrection = false;
+	bool _vsync = true;
+
+	GraphicsState _currentState = {};
+	int _pendingScreenChange = PendingScreenChange::None;
+
+	bool _screenModified = false;	// double/triple buffering only
+	Common::Rect _modifiedScreenRect;	// direct rendering only
+
+	Common::Array<Common::Rect> _modifiedChunkyRects;
+
+	bool _overlayVisible = false;
+	Common::Array<Common::Rect> _modifiedOverlayRects;
+
+	struct Cursor {
+		void update(const Graphics::Surface &screen);
+
+		bool isModified() const {
+			return surfaceChanged || positionChanged;
+		}
+
+		bool visible = false;
+
+		// position
+		Common::Point getPosition() const {
+			return Common::Point(x, y);
+		}
+		void setPosition(int _x, int _y, bool override = false) {
```

Comment:
Names starting with an underscore are reserved for the class variable. Consider using `x_`